Automattic/wp-calypso
# Plans: cleanup the "show fewer features" experiment

Related to https://github.com/Automattic/martech/issues/3535

## Proposed Changes

* This change removes the the experiment added in Automattic/wp-calypso#94044 and ships the "simplified" variant.
* The simplified variant will eventually be the default for all pricing grids, but currently, it will be only active for the `/start` signup flow and the logged-in plans page. More details here: p5uIfZ-ggB-p2#comment-24241

## Why are these changes being made?


* https://github.com/Automattic/martech/issues/3535

## Testing Instructions



* Go to `/start` and `/plans/<site slug>` and confirm that the simplified plans grid is shown.
* Go to `/setup/newsletter/plans` and `/setup/design-first/plans?siteId=<site id>` and confirm that the plans grid does not differ from production.

## Pre-merge Checklist
